### PR TITLE
Backend support for text sharings

### PR DESCRIPTION
**Pull request by swemail**

### CODE REVIEW

File: shared/src/types/Post.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import type {Timestamp} from 'firebase-admin/firestore';
+import {LANGUAGE_TAG} from '../constants/i18n';
+import {UserProfile} from './User';
+
+export type PostFields = {
+  id: string;
+  exerciseId: string;
+  userId?: string;
```
Comment (gewfy) [inline shared/src/types/Post.ts:8]: I guess this is only used for the anonymity switch and not to keep track of your own sharings?

File: functions/src/api/posts/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        {
+          text: 'some new text',
+        },
+      );
+      expect(response.status).toBe(200);
+    });
+  });
+
+  describe('delete', () => {
+    it('should update post with public flag', async () => {
```
Comment (gewfy) [inline functions/src/api/posts/index.spec.ts:140]: Not correct description, right?

**Review by gewfy — COMMENTED** [review]
Hmm I'm not really following this `public` concept. Can you elaborate?

File: shared/src/types/Post.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  language: LANGUAGE_TAG;
+  public: boolean;
```
[the remark this answers is not in the document]
Comment (swemail) [reply shared/src/types/Post.ts:10]: I think I will remove that, we will not send private messages to the server. If a user switches from public to private we delete and if add again if switching back

File: shared/src/types/Post.ts (hunk 1 above)
Comment (swemail) [reply shared/src/types/Post.ts:8]: That was the idea yes but I think I didn't follow the thought through 😄

File: shared/src/types/Post.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (gewfy) [reply shared/src/types/Post.ts]: 👌

File: firestore.indexes.json [hunk 4]
```diff
@@ -63,6 +63,28 @@
...
+          "order": "ASCENDING"
+        },
+        {
+          "fieldPath": "approved",
+          "order": "ASCENDING"
+        },
+        {
+          "fieldPath": "public",
+          "order": "ASCENDING"
+        },
```
Comment (gewfy) [inline firestore.indexes.json:82]: This should be removed, right?

**Review by gewfy — APPROVED** [review]

File: firestore.indexes.json (hunk 4 above)
Comment (swemail) [reply firestore.indexes.json:82]: Yes 😄

**Pull request merged**